src/main/java/service/impl/InputHandler.java

```diff
@@ -0,0 +1,32 @@
+package service.impl;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import service.InputFileHandler;
+
+public class InputHandler implements InputFileHandler {
+    @Override
+    public List<String> readFile(String fileName) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+                new FileReader(fileName))) {
```

Comment:
You are using the FileReader class, which does not support encoding and can cause problems when reading files with special characters. Consider using the InputStreamReader class, which supports encoding. For example, new InputStreamReader(new FileInputStream(fileName), StandardCharsets.UTF_8).